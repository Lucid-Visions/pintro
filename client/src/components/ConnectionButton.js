import React from "react";
import { View, StyleSheet, Text, TouchableOpacity, Image } from "react-native";
import ElevatedView from "react-native-elevated-view";
import PropTypes from "prop-types";

/**
 * Renders an action button for the user's profile.
 * @param {string} name the type of the button.
 * @param {string} context the specific topic related to the company.
 * @param {string} source the company picture path.
 */
const ConnectionButton = ({ name, text, source, onPress, onLongPress, isActive, swipe }) => (
  <ElevatedView style={{... styles.view, backgroundColor: isActive ? "rgba(144,144,144,0.6)" : "white"}} elevation={5}>
    <TouchableOpacity onPressOut={!!swipe.openDirection ? swipe.close : swipe.openRight} onPress={onPress} style={styles.container}>
      <View style={{ flexDirection: "row" }}>
        <Image
          style={styles.image}
          source={
            source
              ? { url: source }
              : require("../assets/empty-profile-picture.png")
          }
        />
        <View style={styles.textContainer}>
          <Text style={styles.title}>{name}</Text>
          <Text style={styles.subtitle}>{text}</Text>
        </View>
      </View>

      <TouchableOpacity
        style={{
          alignItems: "center",
          justifyContent: "center"
        }}
        onPressIn={onLongPress}
      >
        <Image
          style={styles.dragHandler}
          source={require("../assets/dragHandler.png")}
        />
      </TouchableOpacity>
    </TouchableOpacity>
  </ElevatedView>
);

// Type checker
ConnectionButton.propTypes = {
  name: PropTypes.string,
  text: PropTypes.string,
  source: PropTypes.string,
  swipe: PropTypes.any.isRequired
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  view: {
    alignSelf:'center',
    flexDirection: "row",
    backgroundColor: "white",
    borderRadius: 15,
    width: "99%",
    resizeMode: "contain",
    padding: 20,
    marginVertical: 10
  },
  textContainer: {
    flexDirection: "column",
    marginLeft: 15,
    marginTop: 5
  },
  title: {
    fontWeight: "bold",
    fontSize: 16
  },
  subtitle: {
    fontWeight: "normal",
    fontSize: 12,
    color: "grey"
  },
  image: {
    borderRadius: 150 / 2,
    width: 40,
    height: 40,
    overflow: "hidden"
  },
  dragHandler: {
    width: 30,
    height: 35
  }
});

export default ConnectionButton;
