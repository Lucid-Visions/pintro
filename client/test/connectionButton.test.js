import React from 'react';
import renderer from 'react-test-renderer';
import ConnectionButton from '../src/components/ConnectionButton';

test('Button renders correctly', () => {
    const component = renderer.create(<ConnectionButton swipe={(() => null, () => null, () => null)} />).toJSON();
    expect(component).toMatchSnapshot();
});