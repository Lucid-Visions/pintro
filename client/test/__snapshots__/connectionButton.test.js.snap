// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "shadowOffset": Object {
          "height": 3,
        },
        "shadowOpacity": 0.1875,
        "shadowRadius": 2.7,
      },
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 15,
        "flexDirection": "row",
        "marginVertical": 10,
        "padding": 20,
        "resizeMode": "contain",
        "width": "99%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 75,
            "height": 40,
            "overflow": "hidden",
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginLeft": 15,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 12,
              "fontWeight": "normal",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
